src/Validator.php

```diff
@@ -8,21 +8,25 @@
 use Merkeleon\PhpCryptocurrencyAddressValidation\Contracts\Driver;
 use Merkeleon\PhpCryptocurrencyAddressValidation\Enums\CurrencyEnum;
 use Merkeleon\PhpCryptocurrencyAddressValidation\Exception\AddressValidationException;
-use function app;
-use function config;
 
-readonly class Validator implements Contracts\Validator
+class Validator implements Contracts\Validator
 {
+    private const CONFIG_PATH = __DIR__ . '/../config/address_validation.php';
+
+    /**
+     * @param DriverConfig[] $options
+     */
     public function __construct(
-        private string $chain,
-        private array  $options,
-        private bool   $isMainnet = true
+        private readonly string $chain,
+        private readonly bool $isMainnet = true,
+        private ?array $options = null,
     ) {
+        $this->options = $options ?? $this->resolveConfigForCurrency(CurrencyEnum::from($this->chain));
     }
 
-    public static function make(CurrencyEnum $currency): Validator
+    public static function make(CurrencyEnum $currency, bool $isMainnet = true, ?array $config = null): self
```

Comment:
since this factory method copies current construct method I was thinking if it would make sense to remove it. But I left it here for backwards compatibility reasons. Maybe we can docblock it with @deprecated to encourage developers to use constructor instead?